catalog/internal/catalog/performance_metrics.go

```diff
@@ -0,0 +1,89 @@
+package catalog
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/golang/glog"
+	dbmodels "github.com/kubeflow/model-registry/catalog/internal/db/models"
+	"github.com/kubeflow/model-registry/catalog/internal/db/service"
+	model "github.com/kubeflow/model-registry/catalog/pkg/openapi"
+)
+
+// LoadPerformanceMetricsData loads performance metrics data from the specified directory
+// into the database using the catalog model and artifact repositories.
+func LoadPerformanceMetricsData(path []string, modelRepo dbmodels.CatalogModelRepository, metricsArtifactRepo dbmodels.CatalogMetricsArtifactRepository, typeMap map[string]int64) (map[string]*model.CatalogModel, error) {
+	if len(path) == 0 {
+		glog.Info("No performance metrics path provided, skipping performance metrics loading")
+		return nil, nil
+	}
+
+	// Check if path exists
+	for _, p := range path {
+		if _, err := os.Stat(p); os.IsNotExist(err) {
+			glog.Warningf("Performance metrics path %s does not exist, skipping performance metrics loading", p)
+			return nil, nil
+		}
+	}
+
+	glog.Infof("Loading performance metrics data from %s", path)
+
+	// Get the TypeID for CatalogModel from the type map
+	modelTypeIDInt64, exists := typeMap[service.CatalogModelTypeName]
+	if !exists {
+		return nil, fmt.Errorf("CatalogModel type not found in type map")
+	}
+	modelTypeID := int32(modelTypeIDInt64)
+	glog.V(2).Infof("Using catalog model type ID: %d", modelTypeID)
+
+	// Get the TypeID for CatalogMetricsArtifact from the type map
+	metricsArtifactTypeIDInt64, exists := typeMap[service.CatalogMetricsArtifactTypeName]
+	if !exists {
+		return nil, fmt.Errorf("CatalogMetricsArtifact type not found in type map")
+	}
+	metricsArtifactTypeID := int32(metricsArtifactTypeIDInt64)
+	glog.V(2).Infof("Using metrics artifact type ID: %d", metricsArtifactTypeID)
+
+	loadedModels := make(map[string]*model.CatalogModel)
+	processedCount := 0
```

Comment:
We don't necessarily need these to exist in memory, other than to get a count which is logged out later. I figured it is useful, at least for now